_posts/2021-02-09-message-passing.md

```diff
@@ -0,0 +1,188 @@
+---
+layout: post
+title: A few notes on message passing
+tags: BEAM erts jit
+author: John Högberg
+---
+
+Message passing has always been central to Erlang, and while reasonably
+well-documented we've avoided going into too much details to give us more
+freedom when implementing it. There's nothing preventing us from describing it
+in a blog post though, so let's have a closer look!
+
+Erlang processes communicate with each other by sending each other _signals_
+(not to be confused with Unix signals). There are many different kinds and
+_messages_ are just the most common one. Practically everything involving more
+than one process uses signals internally: for example, the `link/1` function is
+implemented by having the involved processes talk back and forth until they've
+agreed on a link.
+
+This helps us avoid a great deal of locks and would make an interesting blog
+post on its own, but for now we only need to keep two things in mind: all
+signals (including messages) are continuously received and handled behind the
+scenes, and they have a [defined order]:
+
+Signals between two processes are guaranteed to arrive in the order they were
```

Comment:
Should we mention something about message delivery guarantees?

Reply:
I'll add a paragraph on distributed sending under "Sending messages"